client/dave_client.go

```diff
@@ -1,20 +1,46 @@
 package client
 
 import (
-	"os"
+	"net/http"
 
-	"github.com/Gamebuildr/Dave/pkg/config"
+	"github.com/Gamebuildr/Dave/pkg/scaler"
 	"github.com/Gamebuildr/Dave/pkg/watcher"
+	"github.com/Gamebuildr/gamebuildr-lumberjack/pkg/logger"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sqs"
 )
 
+// DaveClient scales microservices remotely
 type DaveClient struct {
-	Watcher watcher.AmazonWatcher
+	Watcher watcher.QueueMonitor
+	Log     logger.Log
 }
 
+// Create a new DaveClient
 func (client *DaveClient) Create() {
 	sess := session.Must(session.NewSession())
-	client.Watcher = watcher.AmazonWatcher{Client: sqs.New(sess)}
-	client.Watcher.ReadQueueMessages(os.Getenv(config.gogetaSQSEndpoint))
+	client.Watcher.Queue = watcher.AmazonWatcher{Client: sqs.New(sess)}
+}
+
+// RunClient will read queue messages and send a response to an api endpoint to
+// start up systems if the load is less than the max load
+func (client *DaveClient) RunClient(system *scaler.ScalableSystem, queueUrl string) {
+	count, err := client.Watcher.Queue.ReadQueueMessages(queueUrl)
```

Comment:
Is this an us method or amazon? if it's ours maybe we should call it ReadQueueMessagesCount as it's a tad misleading